testdata/src/main/java/com/hedvig/app/testdata/dashboard/builders/InsuranceDataBuilder.kt

```diff
@@ -32,47 +38,51 @@ class InsuranceDataBuilder(
                 id = "120e9ac9-84b1-4e5d-add1-70a9bad340be",
                 status = InsuranceQuery.Status(
-                    __typename = c.toTypename(),
+                    __typename = c.typename,
                     fragments = InsuranceQuery.Status.Fragments(
                         contractStatusFragment = ContractStatusFragment(
-                            __typename = c.toTypename(),
+                            __typename = c.typename,
                             asPendingStatus = if (c == ContractStatus.PENDING) {
                                 ContractStatusFragment.AsPendingStatus(
-                                    __typename = "",
+                                    __typename = c.typename,
                                     pendingSince = null
                                 )
                             } else {
                                 null
                             },
                             asActiveInFutureStatus = when (c) {
                                 ContractStatus.ACTIVE_IN_FUTURE -> ContractStatusFragment.AsActiveInFutureStatus(
-                                    __typename = "",
+                                    __typename = c.typename,
                                     futureInception = LocalDate.of(2025, 1, 1)
                                 )
                                 ContractStatus.ACTIVE_IN_FUTURE_INVALID ->
                                     ContractStatusFragment.AsActiveInFutureStatus(
-                                        __typename = "",
+                                        __typename = c.typename,
                                         futureInception = null
                                     )
                                 else -> null
                             },
                             asActiveStatus = if (c == ContractStatus.ACTIVE) {
                                 ContractStatusFragment.AsActiveStatus(
-                                    __typename = "",
-                                    pastInception = LocalDate.now(),
-                                    upcomingAgreementChange = ContractStatusFragment.UpcomingAgreementChange(
-                                        newAgreement = ContractStatusFragment.NewAgreement(
-                                            __typename = "",
-                                            asSwedishApartmentAgreement = ContractStatusFragment
-                                                .AsSwedishApartmentAgreement(
-                                                    __typename = "",
-                                                    activeFrom = LocalDate.of(2021, 4, 6)
-                                                )
+                                    __typename = c.typename,
+                                    pastInception = LocalDate.of(2021, 1, 6),
+                                    upcomingAgreementChange = if (showUpcomingAgreement) {
```

Comment:
When constructing these objects, if we have both a normal field and a fragment that are referencing the same thing (upcomingAgreementChange here) they need to be equal. This is because the json string returns this information once, and then the adapter fills both the normal field and the field inside the fragment from that same information in the json string. Before since we were not validating that this object was a proper object that could be generated we were having different data for upcomingAgreementChange in the same response. We can't do that no more so we need to be careful in the future as well with similar cases.